basic_algs/src/main.rs

```diff
@@ -0,0 +1,77 @@
+fn max_element(arr: &[i32; 10]) -> i32 {
+    let mut res = i32::MIN;
+    for x in arr {
+        if *x > res {
+            res = *x;
+        }
+    }
+    res
+}
+
+fn nth_prime(n: u32) -> u32 {
+    let mut primes = Vec::new();
+    for x in 2u32..u32::MAX {
+        let mut is_prime = true;
+        for d in &primes {
+            if x % *d == 0 {
+                is_prime = false;
+                break;
+            }
+            if d * d > x {
+                break;
+            }
+        }
+        if is_prime {
+            primes.push(x);
+            if primes.len() >= n as usize {
+                break;
+            }
+        }
+    }
+    *primes.last().unwrap()
+}
+
+fn find_num(arr: &[i32; 10], needle: &i32) -> Option<usize> {
```

Comment:
Copy-типы лучше принимать по значению:
```suggestion
fn find_num(arr: &[i32; 10], needle: i32) -> Option<usize> {
```
По следующим причинам:
1. Мы избавляемся от разыменований в теле функции, код становится чуть чище
2. Нет необходимости в `&` при вызове функции - также меньше шума.
3. В некоторых случаях передача значения более эффективна, чем передача ссылки, особенно для типов меньше размера указателя. Впрочем, микробенчмарки показывают очень нестабильные результаты.